pkg/kubectl/cmd/logs/logs.go

```diff
@@ -191,9 +193,7 @@ func (o *LogsOptions) ToLogOptions() (*corev1.PodLogOptions, error) {
 		logOptions.SinceSeconds = &sec
 	}
 
-	if len(o.Selector) > 0 && o.Tail == -1 {
-		logOptions.TailLines = &selectorTail
-	} else if o.Tail != -1 {
+	if o.TailSpecified {
```

Comment:
The first condition should be left untouched. You can eventually change it to `!o.TailSpecified`